core/test/control_test.jl

```diff
@@ -11,16 +11,16 @@
 
     # Control input
     pump_control_mapping = p.pump.control_mapping
-    @test pump_control_mapping[(NodeID(:Pump, 4), "off")].flow_rate == 0
-    @test pump_control_mapping[(NodeID(:Pump, 4), "on")].flow_rate == 1.0e-5
-
-    logic_mapping::Dict{Tuple{NodeID, String}, String} = Dict(
-        (NodeID(:DiscreteControl, 5), "TT") => "on",
-        (NodeID(:DiscreteControl, 6), "F") => "active",
-        (NodeID(:DiscreteControl, 5), "TF") => "off",
-        (NodeID(:DiscreteControl, 5), "FF") => "on",
-        (NodeID(:DiscreteControl, 5), "FT") => "off",
-        (NodeID(:DiscreteControl, 6), "T") => "inactive",
+    @test pump_control_mapping[(NodeID(:Pump, 4), "off")].flow_rate_scalar == 0
+    @test pump_control_mapping[(NodeID(:Pump, 4), "on")].flow_rate_scalar == 1.0e-5
+
+    logic_mapping::Dict{Tuple{NodeID, Vector{Bool}}, String} = Dict(
+        (NodeID(:DiscreteControl, 5), [true, true]) => "on",
```

Comment:
You might want to use my suggested string macro to keep the "TT" in tests for readability. Up to you.